internal/js/modules/k6/browser/common/network_manager.go

```diff
@@ -881,10 +881,29 @@ func (m *NetworkManager) AbortRequest(requestID fetch.RequestID, errorReason str
 	return nil
 }
 
-func (m *NetworkManager) ContinueRequest(requestID fetch.RequestID) error {
+func (m *NetworkManager) ContinueRequest(
+	requestID fetch.RequestID,
+	opts ContinueOptions,
+	originalHeaders []HTTPHeader,
+) error {
 	m.logger.Debugf("NetworkManager:ContinueRequest", "continuing request (id: %s)", requestID)
-
 	action := fetch.ContinueRequest(requestID)
+
+	if len(opts.Headers) > 0 {
+		allHeaders := mergeHeaders(originalHeaders, opts.Headers)
+		action = action.WithHeaders(allHeaders)
```

Comment:
Looking at the example in https://playwright.dev/docs/api/class-route#route-continue - if headers are provided it is overriding them. Otherwise it woulnd't be copying the headers ?!?!

There is also some comment at the end for Cookie ... which might actually just be that browsers will do this irregardless   :thinking:

Reply:
Right, I got confused because these are called "overrides" in PW codebase but you're right. Updated in https://github.com/grafana/k6/pull/4971/commits/591b9a84a691cfffb82e09e918724b7b13d863cc

Regarding cookie, they do have a specific filter in their codebase to remove it (https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/server/network.ts#L352-L353) I wasn't sure why to be honest.